packages/story-editor/src/components/panels/design/layer/layer.js

```diff
@@ -352,6 +352,10 @@ function Layer({ element }) {
     })
   );
 
+  useEffect(() => {
+    setNewLayerName(layerName);
+  }, [layerName]);
```

Comment:
With this, I think we can also remove  almost all the other places, where `setNewLayerName` is invoked, as most of them are workarounds to exactly this thing missing.

Reply:
Okay so 

I think we'll still need the `handleChange` and `handleKeyDown`

But we can update `updateLayerName` to be this code below ya?

```javascript
const updateLayerName = () => {
    setRenamableLayer(null);
    const trimmedLayerName = newLayerName.trim();
    if (trimmedLayerName) {
      updateElementById({
        elementId: element.id,
        properties: { layerName: trimmedLayerName },
      });
    }
  };
```

Reply:
Pushed an update for that 

https://github.com/GoogleForCreators/web-stories-wp/pull/11700/commits/c1680cdb393ce3094b03553aed31499fb3db82d3